### PR TITLE
Extract `fail` helper function

### PR DESCRIPTION
**Pull request by jasonkarns**
fail helper prints message to stderr and exits 1

### CODE REVIEW

File: libexec/rbenv [hunk 1]
```diff
@@ -12,21 +12,25 @@ if [ -n "$RBENV_DEBUG" ]; then
   set -x
 fi
 
+fail() {
+  {
```
Comment (mislav) [inline libexec/rbenv]: I usually start the block right on the opening bracket line:

```sh
{ if [ ... ]; ....
  ...
}
```

File: libexec/rbenv [hunk 2]
```diff
@@ -12,21 +12,25 @@ if [ -n "$RBENV_DEBUG" ]; then
...
+    if [ "$#" -eq 0 ]; then cat -
+    else echo -e "$@"
```
Comment (mislav) [inline libexec/rbenv]: Why `-e`?

**Comment by mislav** [comment]
Thanks for the suggestion. I see how this encapsulates a pattern we repeat in the same file a lot; I'm just not sure whether it's really worthwhile to DRY this up. It's really not that hard to redirect to stderr and `exit 1`. But I'll think about this :+1:

File: libexec/rbenv [hunk 3]
```diff
@@ -12,21 +12,25 @@ if [ -n "$RBENV_DEBUG" ]; then
   set -x
 fi
 
+fail() {
```
Comment (mislav) [inline libexec/rbenv]: Perhaps we could call it `abort`?

File: libexec/rbenv [hunk 4]
```diff
@@ -92,10 +93,7 @@ shopt -u nullglob
 command="$1"
 case "$command" in
 "" )
-  { rbenv---version
-    rbenv-help
-  } >&2
-  exit 1
+  { rbenv---version; rbenv-help; } | fail
```
Comment (mislav) [inline libexec/rbenv]: I would split this into separate lines like before

File: libexec/rbenv (hunk 2 above)
Comment (mislav) [inline libexec/rbenv]: How about if a string argument was given, we prefix it with `rbenv: ` always? That way we don't have to repeat `rbenv: ` in each message in this file.

File: libexec/rbenv (hunk 3 above)
Comment (jasonkarns) [inline libexec/rbenv]: :+1:

File: libexec/rbenv (hunk 1 above)
Comment (jasonkarns) [inline libexec/rbenv]: that breaks vim's auto-formatting/indentation for SH files.

```
abort() {
  { if [ "$#" -eq 0 ]; then cat -
  else echo -e "$@"
  fi
} >&2
exit 1
}
```

But can switch if still desired

Comment (mislav) [inline libexec/rbenv]: Yeah, please switch to match the currently prevalent style in rbenv and ruby-build

**Comment by jasonkarns** [comment]
i think i addressed everything

**Comment by mislav** [comment]
Thanks. How about squashing this

**Comment by jasonkarns** [comment]
squashed

**Comment by mislav** [comment]
Thanks!

**Pull request merged**